### ISSUE TITLE
Enhance `get_arp_table` and `get_mac_address_table` with filtering arguments

### ISSUE THREAD

**Issue by kbirkeland** [napalm-automation/napalm-base#217]
I have to deal with some rather large ARP tables (>20k), so it would be nice to have some filtering built into the API.

I propose the following arguments for `get_arp_table`:

 * `ip`
 * `mac`
 * `interface`

and the following arguments for `get_mac_address_table`:

 * `mac`
 * `interface`
 * `vlan`

**Comment by mirceaulinic** [comment]
I am going to approve this request, it makes sense to me. Let's see what @dbarrosop and @ktbyers think about that as well.

**Comment by ktbyers** [comment]
Yes, I think this is a good idea.

**Comment by dbarrosop** [comment]
+1

**Comment by mirceaulinic** [comment]
Hi @kbirkeland - this API change is approved. I have created the issues for each driver.
As any API change, we must synchronise all drivers and apply the changed concomitantly. Would you be able to help with this?

Thanks!

**Comment by kbirkeland** [comment]
Yeah, I can help with this, but I don't have access to a VM for all the driver types.  What's the best course of action for those?  Just raise NotImplementedError?

**Comment by mirceaulinic** [comment]
Hi @kbirkeland - no problem about the platforms you don't have, please contribute as much as you are able to. Thanks!

**Comment by icmp-echo** [comment]
@mirceaulinic I have simple question Sir, if I wanna pull get_vlan just like get_interfaces or get_arp_table, 

how do i do that? where can i create or modify an existing call to do so?

Again I cant thank you enough for the Napalm and much more awesome things we have here today.
